quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/infinispan/jgroups/JGroupsUtil.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2025 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.quarkus.runtime.storage.infinispan.jgroups;
+
+import org.infinispan.commons.configuration.attributes.Attribute;
+import org.infinispan.configuration.global.TransportConfigurationBuilder;
+import org.infinispan.configuration.parsing.ConfigurationBuilderHolder;
+import org.jgroups.protocols.TCP_NIO2;
+import org.jgroups.protocols.UDP;
+import org.keycloak.quarkus.runtime.storage.infinispan.CacheManagerFactory;
+
+import static org.infinispan.configuration.global.TransportConfiguration.STACK;
+
+public final class JGroupsUtil {
+
+    private JGroupsUtil() {
+    }
+
+    public static TransportConfigurationBuilder transportOf(ConfigurationBuilderHolder holder) {
+        return holder.getGlobalConfigurationBuilder().transport();
+    }
+
+    public static Attribute<String> transportStackOf(ConfigurationBuilderHolder holder) {
+        var transport = transportOf(holder);
+        assert transport != null;
+        return transport.attributes().attribute(STACK);
+    }
+
+    public static void warnDeprecatedStack(ConfigurationBuilderHolder holder) {
+        var stackName = transportStackOf(holder).get();
+        switch (stackName) {
+            case "jdbc-ping-udp":
+            case "tcp":
+            case "udp":
+            case "azure":
+            case "ec2":
+            case "google":
+                CacheManagerFactory.logger.warnf("Stack '%s' is deprecated. We recommend to use 'jdbc-ping' instead", stackName);
+        }
```

Comment:
Wouldn't be better to use `CachingOptions.Stack` ?

Reply:
the `CachingOptions.Stack` is copied to the Infinispan's configuration; it should be fine.